src/esn.calendar.libs/app/services/shells/calendar-shell.js

```diff
@@ -801,6 +802,37 @@ require('./valarm-shell.js');
       });
     }
 
+    /**
+     * Returns the differences between this and the event object to compare to.
+     * @param  {CalendarShell} comparedTo an event
+     *
+     * @return {Array} an array containing the keys of differences between events.
+     */
+    function difference(comparedTo) {
+      const differences = [];
+      const self = this;
+
+      CAL_EVENT_MODIFY_COMPARE_KEYS.forEach(key => {
+        if (['start', 'end'].includes(key)) {
+          // check if both are not undefined and one of them is undefined
+          if (!(!self[key] && !comparedTo[key]) && (!self[key] || !comparedTo[key])) {
+            differences.push(key);
+          } else if (self[key] !== comparedTo[key] && (self[key]._isAMomentObject && comparedTo[key]._isAMomentObject && !self[key].isSame(comparedTo[key]))) {
+            differences.push(key);
+          }
+        } else if (['rrule', 'alarm'].includes(key)) {
+          // Check if both are not undefined and one of them is undefined or one of them is not equal to the other.
+          if (!(!self[key] && !comparedTo[key]) && ((!self[key] || !comparedTo[key]) || !self[key].equals(comparedTo[key]))) {
+            differences.push(key);
+          }
+        } else if (!angular.equals(self[key], comparedTo[key])) { // any regular field gets checked here
+          differences.push(key);
+        }
+      });
+
+      return differences;
```

Comment:
```suggestion
      const self = this;

      return CAL_EVENT_MODIFY_COMPARE_KEYS.map(key => {
        if (!angular.equals(self[key], comparedTo[key])) { // any regular field gets checked here
          return key;
        }

        if (['rrule', 'alarm'].includes(key) && ((self[key] || comparedTo[key]) && ((!self[key] || !comparedTo[key]) || !self[key].equals(comparedTo[key]))) {
          return key;
        }

        if (!['start', 'end'].includes(key)) return;

        if (!(!self[key] && !comparedTo[key]) && (!self[key] || !comparedTo[key])) {
          return key;
        }

        if (self[key] !== comparedTo[key] && (self[key]._isAMomentObject && comparedTo[key]._isAMomentObject && !self[key].isSame(comparedTo[key]))) {
          return key;
        }
      }).filter(Boolean);
```

Reply:
There, no nested if-elses. Hopefully I didn't mess anything up :trollface:.

Reply:
i like the idea.